pkg/server/admin.go

```diff
@@ -1436,10 +1436,20 @@ func (s *adminServer) Events(
 
 	userName, err := s.requireAdminUser(ctx)
 	if err != nil {
-		// NB: not using serverError() here since the priv checker
-		// already returns a proper gRPC error status.
-		return nil, err
+		if !errors.Is(err, errRequiresAdmin) {
+			return nil, err
+		}
+		viewActivityPermErr := s.adminPrivilegeChecker.requireViewActivityOrViewActivityRedactedPermission(ctx)
```

Comment:
This is not sound error handling.

There is a difference between "An error was encountered while retrieving the privilege metadata" (and that error should be propagated as-is to the client, and logged) and "The privileges say NO" (authorization error, propagated as a synthetic condition to the client).

Generally, authorization methods should have a `bool, error` return. Errors always interrupt the control flow; the bool can be used to make authz decisions.